float64 poseX
float64 poseY
float64 poseZ